src/main/java/network/brightspots/rcv/CommonDataFormatReader.java

```diff
@@ -85,11 +85,38 @@ CVRContest getCvrContestXml(CVR cvr, Contest contestToTabulate) {
 
   void parseCvrFile(List<CastVoteRecord> castVoteRecords)
       throws UnrecognizedCandidatesException, IOException, CvrParseException {
-    if (filePath.endsWith(".xml")) {
-      parseXml(castVoteRecords);
-    } else if (filePath.endsWith(".json")) {
-      parseJson(castVoteRecords);
+    try {
+      if (filePath.endsWith(".xml")) {
+        parseXml(castVoteRecords);
+      } else if (filePath.endsWith(".json")) {
+        parseJson(castVoteRecords);
+      }
+    } catch (CvrParseException e) {
+      throw e;
+    } catch (UnrecognizedCandidatesException e) {
+      throw e;
+    } catch (Exception e) {
+      Logger.severe("Unknown error. Cannot load file.");
+      throw new CvrParseException();
+    }
+  }
+
+  /**
+   * Sets map[key] = value if map does not contain key already.
+   * Throws CvrParseException if it does.
+   *
+   * @param map The map to modify
+   * @param humanReadableName Used as part of the error message if the key is already in the map
```

Comment:
Nit: Let's just call `humanReadableName` => `keyName` instead, and the description can say it's a "human-readable name of the key, used to make the error message more useful".